### PR TITLE
feat: optimisation - memory pooling, FfiHandles

### PR DESCRIPTION
**Pull request by NickKhalow**
- remove FfiHandles from protocol
- IntPtr for FfiHandles
- memory pooling
- inner message pooling per request

### CODE REVIEW

File: Runtime/Scripts/Internal/FFIClients/Pools/Memory/MemoryPool.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Collections.Generic;
+
+namespace LiveKit.Internal.FFIClients.Pools.Memory
+{
+    public class MemoryPool : IMemoryPool
```
Comment (mikhail-dcl) [inline Runtime/Scripts/Internal/FFIClients/Pools/Memory/MemoryPool.cs:6]: This class should be thread-safe as we will be using the shared `instance` of `FFIBridge` to send requests unless we somehow can guarantee that we invoke it from a single background thread only (but it contradicts the singleton principle).
I propose you rely on `ArrayPool` - it already contains the synchronization on the chunk basis out of the box.

**Review by mikhail-dcl — CHANGES_REQUESTED** [review]
This is exactly what we needed 💪 
Please check thread safety: it also applies to `MultiPool` as we use it from the shared instance of `FFIBridge`

File: Runtime/Scripts/Internal/FFIClients/Pools/Memory/MemoryPool.cs (hunk 1 above)
Comment (NickKhalow) [reply Runtime/Scripts/Internal/FFIClients/Pools/Memory/MemoryPool.cs:6]: I think it will be great to rid off static Singleton

WDYT?

Comment (mikhail-dcl) [reply Runtime/Scripts/Internal/FFIClients/Pools/Memory/MemoryPool.cs]: We can get rid of it but then we will need to have an instance of the FFIClient anyway and I believe it still will be in a single instance for multiple rooms and streams

**Review by mikhail-dcl — APPROVED** [review]
great, thanks

**Pull request merged**